### ISSUE TITLE
Generic.Functions.CallTimePassByReference.NotAllowed false positive when short array syntax

### ISSUE THREAD

**Issue by legoktm** [squizlabs/PHP_CodeSniffer#897]
The following code will (wrongly) trigger the sniff:
```
Hooks::run( 'SecondaryDataUpdates', [ $title, $old, $recursive, $parserOutput, &$updates ] );
```

This code is fine though:
```
Hooks::run( 'SecondaryDataUpdates', array( $title, $old, $recursive, $parserOutput, &$updates ) );
```

We observed this while mass-migrating MediaWiki to []-style arrays usinb phpcbf :)

Downstream bug: https://phabricator.wikimedia.org/T127163

**Comment by gsherwood** [comment]
Thanks for reporting this issue.
